extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,470 @@
+<?php
+
+class RenameIPProcess {
+
+	const RENAME_TAG = 'renamed_to';
+	const PROCESS_TAG = 'rename_in_progress';
+	const PHALANX_BLOCK_TAG = 'phalanx_block_id';
+
+	const MAX_ROWS_PER_QUERY = 500;
+
+	const LOG_STANDARD = 'standard';
+	const LOG_BATCH_TASK = 'task';
+	const LOG_OUTPUT = 'output';
+
+	// Define what needs changing in core MW tables
+	/*
+	 * Task definition format:
+	 *	 'table' => (string) table name
+	 *   'userid_column' => (string) column name with user ID or null if none
+	 *   'username_column' => (string) column name with user name
+	 *   'conds' => (array) additional conditions for the query
+	 */
+	/**
+	 * Stores the predefined tasks to do for global (wikicities) database.
+	 * Here should be mentioned all core tables not connected to any extension.
+	 *
+	 * @var $mGlobalDefaults array
+	 */
+	static private $mGlobalDefaults = array(
+		// user table is processed separately
+//		array( 'table' => '...', 'userid_column' => '...', 'username_column' => '...' ),
+	);
+
+	/**
+	 * Stores the predefined tasks to do for every local wiki database for IP addresses.
+	 * Here should be mentioned all core tables not connected to any extension.
+	 *
+	 * @var $mLocalIpDefaults array
+	 */
+	static private $mLocalIpDefaults = [
+		[ 'table' => 'archive', 'userid_column' => 'ar_user', 'username_column' => 'ar_user_text' ],
+		[ 'table' => 'filearchive', 'userid_column' => 'fa_user', 'username_column' => 'fa_user_text' ],
+		[ 'table' => 'ipblocks', 'userid_column' => 'ipb_user', 'username_column' => 'ipb_address' ],
+		[ 'table' => 'recentchanges', 'userid_column' => 'rc_user', 'username_column' => 'rc_user_text' ],
+		[ 'table' => 'revision', 'userid_column' => 'rev_user', 'username_column' => 'rev_user_text' ],
+	];
+
+	private $mRequestData = null;
+	private $mActionConfirmed = false;
+
+	private $mOldIPAddress = '';
+	private $mNewIPAddress = '';
+	private $mUserId = 0;
+	private $mFakeUserId = 0;
+	private $mPhalanxBlockId = 0;
+	private $mRequestorId = 0;
+	private $mRequestorName = '';
+	private $mReason = null;
+	private $mNotifyUser;
+
+	private $mErrors = array();
+	private $mWarnings = array();
+
+	private $mInternalLog = '';
+
+	private $mLogDestinations = array(
+		array( self::LOG_STANDARD, null ),
+	);
+	/**
+	 *
+	 * @var BatchTask
+	 */
+	private $mLogTask = null;
+
+	/**
+	 * @var string
+	 */
+	private $mUserRenameTaskId = null;
+
+	/**
+	 * Creates new rename user process
+	 *
+	 * @param string $oldUsername Old username
+	 * @param string $newUsername New username
+	 * @param bool $confirmed Has the user confirmed all the warnings he got?
+	 * @param string $reason
+	 * @param bool $notifyUser Whether to notify the renamed user
+	 */
+	public function __construct( $oldIPAddress, $newIPAddress, $confirmed = false, $reason = null ) {
+		global $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app